processor/t60_processor.py

```diff
@@ -26,75 +31,158 @@ def extract_s3_info(object_url: str) -> [str, str]:
+        self.video_path = environ.get('VIDEO_PATH', 'https://s3.amazonaws.com/ai.powern.website.assets/cpms/FWH/360Slam/video.mp4')
+        self.camera_path = environ.get('CAMERA_PATH','https://s3.amazonaws.com/ai.powern.website.assets/cpms/FWH/360Slam/equirectangular.yaml')
+        self.output_path = 'messagepack.msg'
+        self.bucket = ''
+        self.parent_key = ''
+        self.dest_path = './360Images/'
+        self.coordinates_file = 'coordinates.json'
 
     def localize(self):
         '''
             Run the stella-vslam localizer to extract the map information
+            --auto-term end is required to complete process in the background
         '''
         subprocess.run([
-            './run_video_slam', '-v', '/tmp/data/orb_vocab.fbow', '-m', self.video_path, '-c', self.camera_path, '--frame-skip', self.skip_frame, '--no-sleep',
-            '--map-db-in', self.output_path
-        ])
+                './run_video_slam', '-v', './orb_vocab.fbow', '-m', self.video_local_path, '-c', self.camera_local_path, '--frame-skip', str(self.skip_frame), '--no-sleep',
+                '--map-db-out', self.output_path, '--auto-term'
+            ])
+    def extract_frames_and_coordinates(self):
+        '''
+            Extracts the msgpack from the output path
+        '''
+        service = FrameCoordinatesExtractor(self.output_path, self.video_local_path, self.dest_path) 
+        service.generate_frames_from_msg()
+        service.generate_final_coordinates_json()
+        
+        
+    def set_bucket_and_parent_key(self):
+        '''
+            Sets the object and parent key for the output path
+        '''
+        self.bucket, object_key = Utils.extract_s3_info(self.video_path)
+        self.parent_key = Utils.extract_parent_key(object_key)
+
 
     def download(self, source_url: str):
         """
-            Downloads the file object from Source (s3) and creates a temporary file
+        Downloads the file object from Source (s3) and creates a temporary file
         """
-        bucket, object_key = Utils.extract_s3_info(
-            source_url.get("s3_path"))
-
+        if isinstance(source_url, dict):
+            bucket, object_key = Utils.extract_s3_info(source_url.get("s3_path"))
+        else:
+            # Handle the case when source_url is a string
+            bucket, object_key = Utils.extract_s3_info(source_url)
         temp_file = tempfile.NamedTemporaryFile(delete=False)
-
         self._S3_READ_CLIENT.download_fileobj(
             Bucket=bucket, Key=object_key, Fileobj=temp_file)
-
         return temp_file.name
 
+
     def fetch_data(self):
         '''
             Downloads the source data
         '''
-
         self.camera_local_path = self.download(self.camera_path)
-        self.video_local_path = self.download(self.video_local_path)
-
+        self.video_local_path = self.download(self.video_path)
+        print("Downloaded the source data")
+        
+        
     def upload(self):
```

Comment:
can you use multi threading here, the image frame count can easily get big. 

```python
import concurrent.futures
import boto3

# Function to upload a single file to S3
def upload_file(bucket_name, file_name):
    s3 = boto3.client('s3')
    s3.upload_file(file_name, bucket_name, file_name)

# List of files to upload (example: 100 files)
files_to_upload = ['file{}.txt'.format(i) for i in range(1, 101)]

# S3 bucket name
bucket_name = 'your-bucket-name'

# Number of files to upload in each batch
batch_size = 10  # Adjust as needed

# Number of concurrent threads
num_threads = 5  # Adjust as needed

# Use ThreadPoolExecutor to upload files concurrently
with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as executor:
    # Divide files into batches
    batches = [files_to_upload[i:i+batch_size] for i in range(0, len(files_to_upload), batch_size)]

    # Upload each batch of files concurrently
    for batch in batches:
        # Submit upload tasks for each file in the batch
        futures = [executor.submit(upload_file, bucket_name, file_name) for file_name in batch]

        # Wait for all tasks in the batch to complete
        for future in concurrent.futures.as_completed(futures):
            try:
                future.result()  # Get the result of the task
            except Exception as e:
                print(f'Error uploading file: {e}')

```

Please use this as reference, for our current ecs cluster the max threads we can leverage is 2 I think or 3 lets go with 2 with a batch size of 10